### ISSUE TITLE
okcoin international flag not working with exchange factory

### ISSUE THREAD

**Issue by douggie** [timmolter/XChange#860]
when creating an okcoin exchange via the following method, the url is set as okcoin.cn even though the spec as Use_Intl=true.

com.xeiam.xchange.Exchange exchange = ExchangeFactory.INSTANCE.createExchange(spec);

It seems that si.mazi.rescu.RestInvocationHandler is instatiated with https://www.okcoin.cn/api rather than .com

**Comment by jheusser** [comment]
How do you exactly invoke it and how do you create the spec? The following snippet below works for me.

```
   ExchangeSpecification exSpec = new ExchangeSpecification(OkCoinExchange.class);
   exSpec.setExchangeSpecificParameters(exchangeSpecificMap);
...
    Exchange exchange = ExchangeFactory.INSTANCE.createExchange(exSpec);
```

**Comment by glabmoris** [comment]
Same here

This gets the Chinese exchange
```
ExchangeSpecification exSpec = new ExchangeSpecification(OkCoinExchange.class);
exSpec.setExchangeSpecificParametersItem("Use_Intl", true);
```

**Comment by jheusser** [comment]
@glabmoris I tested this with latest snapshot just now and it does work for me. Which version are you using?

Due to latest changes introduced in 3.0.0 you must however create the `Exchange` via the `createExchange(ExchangeSpecification exchangeSpecification)` method, the other factory method does not work (e.g. creating an exchange and modifying the exchange specification).

**Comment by glabmoris** [comment]
I get the error on 3.0.1-SNAPSHOT. Reverting to 2.2.0-SNAPSHOT or 2.1.0-SNAPSHOT solves the issue.

**Comment by jheusser** [comment]
@glabmoris Some of the exchange specific params for OkCoin have been refactored which have an effect I don't fully understand (how the exch. spec. is applied is generally a bit weird in XChange). Will have to look into it more. Could you provide me with a compile-able test class and I can figure out what's wrong?

**Comment by glabmoris** [comment]
This should do the trick

```
package com.xeiam.xchange.examples.okcoin.marketdata;

import java.io.IOException;
import java.util.List;

import com.xeiam.xchange.Exchange;
import com.xeiam.xchange.ExchangeFactory;
import com.xeiam.xchange.ExchangeSpecification;
import com.xeiam.xchange.currency.CurrencyPair;
import com.xeiam.xchange.okcoin.OkCoinExchange;
import com.xeiam.xchange.service.polling.marketdata.PollingMarketDataService;

public class OkCoinCurrencyPairsDemo {

  public static void main(String[] args) throws IOException {

    ExchangeSpecification exSpec = new ExchangeSpecification(OkCoinExchange.class);

    // flag to set Use_Intl (USD) or China (default)
    exSpec.setApiKey("XXXXX");
    exSpec.setSecretKey("YYYYY");
    exSpec.setExchangeSpecificParametersItem("Use_Intl", true);
    Exchange okcoinExchange = ExchangeFactory.INSTANCE.createExchange(exSpec);

    generic(okcoinExchange);
  }

  private static void generic(Exchange okcoinExchange) throws IOException {

    // Interested in the public polling market data feed (no authentication)
    PollingMarketDataService marketDataService = okcoinExchange.getPollingMarketDataService();
    List<CurrencyPair> currencyPairs = marketDataService.getExchangeSymbols();
    for (CurrencyPair currencyPair : currencyPairs) {
      Ticker t = exchange.getPollingMarketDataService().getTicker(cp);
      System.out.println(cp.baseSymbol + " -> " + cp.counterSymbol +  " : " + t.getBid().doubleValue() + "  <-- value is in CNY");
    }
  }

}
```

**Comment by jheusser** [comment]
@glabmoris fixed #863
